src/content/app/tools/blast/state/blast-api/blastApiSlice.ts

```diff
@@ -19,14 +19,16 @@ import type { BaseQueryFn } from '@reduxjs/toolkit/dist/query/baseQueryTypes';
 
 import restApiSlice from 'src/shared/state/api-slices/restSlice';
 
+import { toFasta } from 'src/shared/helpers/formatters/fastaFormatter';
+
 import type { BlastSettingsConfig } from 'src/content/app/tools/blast/types/blastSettings';
 import type { Species } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
 import type { BlastSubmission } from '../blast-results/blastResultsSlice';
 import type { BlastJobResultResponse } from 'src/content/app/tools/blast/types/blastJob';
 
 export type BlastSubmissionPayload = {
   species: Species[];
-  sequences: { id: number; value: string }[];
+  sequences: { id: number; header?: string; value: string }[];
```

Comment:
`sequences` is defined inside multiple types. Can't it be its own type and reused instead?

Reply:
we do have the following:

`export type ParsedInputSequence = {
  header?: string;
  value: string;
};`

Note the extra `id` field in the above sequences. This is needed once the sequences are submitted. I can still use the `ParsedInputSequence` type above but How do i include it with the sequences above?

Reply:
We can create a new type using `ParsedInputSequence`:

```
type BlastInputSequence = ParsedInputSequence & { id: number; };
```

Reply:
Regarding the name of the proposed new type. As the code is written (see `BlastJobSubmit`, line 93), sequences only acquire the `id` field at the moment of submission. Maybe call the type `StoredInputSequence` or `SubmittedInputSequence`?

Reply:
what about just `SubmittedSequence`